src/main.rs

```diff
@@ -18,26 +18,167 @@
+    Ok(warp::reply::with_status(
+        "Payload received",
+        warp::http::StatusCode::OK,
+    ))
+}
+
+fn get_credentials_bytes(listen_addr: &str) -> (Vec<u8>, Vec<u8>) {
+    let (key, cert) = match KEY_SOURCE {
+        "file-system" => (get_key_bytes_fs(), get_cert_bytes_fs()),
+        "generate" => (generate_credentials(&listen_addr)),
+        //no match!
+        _ => panic!("No match for credentials source"),
+    };
+    (key, cert)
+}
+
+//implementation for file system
+fn get_cert_bytes_fs() -> Vec<u8> {
+    let in_path = Path::new("key-material/server.crt");
+    let in_contents: Vec<u8>;
+    in_contents = match std::fs::read(in_path) {
+        Ok(in_contents) => {
+            println!("Contents = of {} bytes", &in_contents.len());
+            in_contents
+        }
+        Err(_) => {
+            println!("Failed to read from file");
+            panic!("We have no data to use");
+        }
+    };
+    in_contents
+}
+
+//implementation for file system
+fn get_key_bytes_fs() -> Vec<u8> {
+    println!("Generating server key (PEM)");
+    let in_path = Path::new("key-material/server.key");
+    let in_contents: Vec<u8>;
+    in_contents = match std::fs::read(in_path) {
+        Ok(in_contents) => {
+            println!("Contents = of {} bytes", &in_contents.len());
+            in_contents
+        }
+        Err(_) => {
+            println!("Failed to read from file");
+            panic!("We have no data to use");
+        }
+    };
+    in_contents
+}
+
+//TODO - this is vital code, and needs to be carefully audited!
+fn generate_credentials(listen_addr: &str) -> (Vec<u8>, Vec<u8>) {
+    let key = Rsa::generate(2048).unwrap();
+    let pkey = PKey::from_rsa(key.clone()).unwrap();
+
+    println!(
+        "Should create a certificate for {}, but using hard-coded 127.0.0.1 instead",
+        &listen_addr
+    );
+
+    let mut x509_name = openssl::x509::X509NameBuilder::new().unwrap();
+    x509_name.append_entry_by_text("C", "GB").unwrap();
+    x509_name.append_entry_by_text("O", "enarx-test").unwrap();
+    //FIXME - we should use &listen-addr, but this fails
+    //x509_name.append_entry_by_text("CN", &listen_addr).unwrap();
+    x509_name.append_entry_by_text("CN", "127.0.0.1").unwrap();
+    let x509_name = x509_name.build();
+
+    let mut x509_builder = openssl::x509::X509::builder().unwrap();
+    if let Err(e) = x509_builder.set_not_before(&Asn1Time::days_from_now(0).unwrap()) {
+        panic!("Problem creating cert {}", e)
+    }
+    if let Err(e) = x509_builder.set_not_after(&Asn1Time::days_from_now(7).unwrap()) {
+        panic!("Problem creating cert {}", e)
+    }
+
+    x509_builder.set_subject_name(&x509_name).unwrap();
+    x509_builder.set_pubkey(&pkey).unwrap();
+    x509_builder.sign(&pkey, MessageDigest::sha256()).unwrap();
+    let certificate = x509_builder.build();
+
+    (
+        key.private_key_to_pem().unwrap(),
+        certificate.to_pem().unwrap(),
+    )
+}
```

Comment:
Use something other than openssl in this function.

Reply:
(We already have a crypto stack via `warp`.)

Reply:
See comment above.